src/io/flutter/dart/DartPlugin.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.dart;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManager;
+import com.intellij.openapi.util.Version;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Provides access to the Dart Plugin.
+ */
+public class DartPlugin {
+
+  /**
+   * Tracks the minimum required Dart Plugin version.
+   */
+  private static final String MINIMUM_REQUIRED_PLUGIN_VERSION = "162.2233";
+  private static final Version MINIMUM_VERSION = Version.parseVersion(MINIMUM_REQUIRED_PLUGIN_VERSION);
+
+  private static final DartPlugin INSTANCE = new DartPlugin();
+
+  private Version myVersion;
+
+  public static DartPlugin getInstance() {
+    return INSTANCE;
+  }
+
+  private IdeaPluginDescriptor getIdeaPluginDescriptor() {
```

Comment:
It's funny, the reason I even made this a singleton rather than just having a bunch of static methods was with an eye towards testing and I don't think a private constructor plays nicely with that but now you've got me wondering...